frontend/packages/dev-console/src/components/monitoring/metrics/QueryInput.tsx

```diff
@@ -0,0 +1,261 @@
+import * as React from 'react';
+import * as fuzzy from 'fuzzysearch';
+import * as _ from 'lodash';
+import { useTranslation } from 'react-i18next';
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
+import { useDispatch, useSelector } from 'react-redux';
+import { queryBrowserPatchQuery, queryBrowserRunQueries } from '@console/internal/actions/observe';
+import { RootState } from '@console/internal/redux';
+import CloseButton from '@console/shared/src/components/close-button';
+
+type QueryInputProps = {
+  index: number;
+  namespace?: string;
+};
+
+const operators = [
```

Comment:
perhaps you could use the terms coming from `codemirror-promql` available here: https://github.com/prometheus/codemirror-promql/blob/main/src/complete/promql.terms.ts.

Like that you will avoid a desync between these lists and the one supported by Promql

Reply:
I'm probably missing some context, but I would be interesting to know why need to redo the autocompletion / suggestion :).
Hopefully this is something already handled by codemirror (with fuzzy filter and so on)

Reply:
Yes we are using indeed the autocompletion and operators from `codemirror-promql` in the new component. This is the old implementation and I just extracted it away as is being used elsewhere.